paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -216,72 +438,305 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+          "Conv input dimension and filter dimension should be the same.");
+      output->ShareDataWith(*residual_param);
+      if (residual_dt == mkldnn::memory::data_type::u8) {
+        uint8_t* output_data = output->mutable_data<uint8_t>(ctx.GetPlace());
+        mkldnninfo->dst_memory_p =
+            mkldnninfo->handler->AcquireDstMemoryFromPrimitive(
+                to_void_cast<uint8_t>(output_data));
+      } else {
+        int8_t* output_data = output->mutable_data<int8_t>(ctx.GetPlace());
+        mkldnninfo->dst_memory_p =
+            mkldnninfo->handler->AcquireDstMemoryFromPrimitive(
+                to_void_cast<int8_t>(output_data));
+      }
+    } else if (!mkldnninfo->force_fp32_output) {
+      if (mkldnninfo->fuse_relu) {
+        uint8_t* output_data = output->mutable_data<uint8_t>(
+            ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+            mkldnninfo->handler->GetDstMemorySize());
+        mkldnninfo->dst_memory_p =
+            mkldnninfo->handler->AcquireDstMemoryFromPrimitive(
+                to_void_cast<uint8_t>(output_data));
+      } else {
+        int8_t* output_data = output->mutable_data<int8_t>(
+            ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+            mkldnninfo->handler->GetDstMemorySize());
+        mkldnninfo->dst_memory_p =
+            mkldnninfo->handler->AcquireDstMemoryFromPrimitive(
+                to_void_cast<int8_t>(output_data));
+      }
     } else {
-      conv_p = handler.AcquireConvolution(src_memory_p, weights_memory_p,
-                                          dst_memory_p);
+      float* output_data = output->mutable_data<float>(
+          ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+          mkldnninfo->handler->GetDstMemorySize());
+      mkldnninfo->dst_memory_p =
+          mkldnninfo->handler->AcquireDstMemoryFromPrimitive(
+              to_void_cast<float>(output_data));
     }
 
-    // push primitive to stream and wait until it's executed
-    pipeline.push_back(*conv_p);
-    stream(stream::kind::eager).submit(pipeline).wait();
+    // create convolution op primitive
+    std::vector<float> scale_bias_data;
+    auto scale_bias_key = *mkldnninfo->key + "@scale_bias";
+    if (bias) {
+      const float* bias_data = bias->data<float>();
+      auto user_bias_md = platform::MKLDNNMemDesc(
+          {bias_tz}, platform::MKLDNNGetDataType<float>(), memory::format::x);
+      auto user_bias_memory_p = mkldnninfo->handler->AcquireBiasMemory(
+          user_bias_md, to_void_cast<float>(bias_data));
+      std::shared_ptr<mkldnn::memory> bias_memory_p;
+      int mask_reorder = is_multi_channel ? 1 << 0 : 1;
+      int count =
+          is_multi_channel
+              ? (convinfo->g > 1
+                     ? (*convinfo->weights_tz)[1] * (*convinfo->weights_tz)[0]
+                     : (*convinfo->weights_tz)[0])
+              : 1;
+      scale_bias_data.resize(count);
```

Comment:
`scale_bias_data` Could be just created here in a form of:
```C++
std::vector<float> scale_bias_data(count);
```

Instead of being created before the `if (bias)`, especially since nothing apart from this branch uses it.